modules/apps/user/user-test/src/testIntegration/java/com/liferay/user/service/test/UserServiceTest.java

```diff
@@ -96,6 +97,57 @@ public void testAddInvalidOrganizationUsers() throws Exception {
 		}
 	}
 
+	@Test
+	public void testAddUserToOrgWithoutPermission() throws Exception {
+		PermissionChecker originalPermissionChecker =
+			PermissionThreadLocal.getPermissionChecker();
+
+		try {
+			Role role = RoleTestUtil.addRole(RoleConstants.TYPE_REGULAR);
+
+			RoleTestUtil.addResourcePermission(
+				role, Organization.class.getName(),
+				ResourceConstants.SCOPE_COMPANY,
+				String.valueOf(TestPropsValues.getCompanyId()),
+				ActionKeys.ASSIGN_MEMBERS);
+			RoleTestUtil.addResourcePermission(
+				role, User.class.getName(), ResourceConstants.SCOPE_COMPANY,
+				String.valueOf(TestPropsValues.getCompanyId()),
+				ActionKeys.UPDATE);
+
+			User user1 = UserTestUtil.addUser();
+			User user2 = UserTestUtil.addUser();
+
+			Organization organization = OrganizationTestUtil.addOrganization();
+
+			PermissionThreadLocal.setPermissionChecker(
+				PermissionCheckerFactoryUtil.create(user1));
+
+			_userLocalService.addRoleUser(role.getRoleId(), user1.getUserId());
```

Comment:
is it correct to do this here? (after changing the permissionChecker)

Reply:
and could you please check if we can use ContextUserReplace for this test?

Reply:
I updated it to use ContextUserReplace